### PR TITLE
ci: Notify QA about infra failures

### PR DESCRIPTION
**Pull request by def-**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by def-** [comment]
@nrainer-materialize This instantly found problems: https://buildkite.com/materialize/test/builds/86538
Apparently we have network errors writing to test analytics quite often?! Pretty shocking

**Comment by nrainer-materialize** [comment]
This seems pretty verbose. Can we at least collapse some part of the message (e.g., the SQL code)?
See also: https://github.com/MaterializeInc/materialize/pull/28107/files#diff-76ee130fa567ca59a2f6ca9fc7c4fea4edafc06fcf050752d19783374006f834R202-R213

**Comment by nrainer-materialize** [comment]
We might also want to consider to conduct a retry after a network error.

**Comment by def-** [comment]
> This seems pretty verbose.

I switched to Slack now

> We might also want to consider to conduct a retry after a network error.

There is a danger that we can cause timeouts by doing that, so I'd suggest only retrying for 1-2 min max (could also check for `BUILDKITE_TIMEOUT` and record the start time in a hook. I think it might also be interesting for us to understand why we see these timeouts first, before adding measures.

File: misc/python/materialize/buildkite.py [hunk 1]
```diff
@@ -201,6 +203,23 @@ def _upload_shard_info_metadata(items: list[str]) -> None:
     )
 
 
+def add_failure_for_qa_team(failure: str) -> None:
```
Comment (nrainer-materialize) [inline misc/python/materialize/buildkite.py:206]: Optional: `notify_qa_team_on_slack_about_failure`?

**Review by nrainer-materialize — APPROVED** [review]

File: misc/python/materialize/buildkite.py (hunk 1 above)
Comment (def-) [reply misc/python/materialize/buildkite.py:206]: I think using Slack is an implementation detail, we could also switch to some other mechanism.

**Pull request merged**